src/metadata.js

```diff
@@ -17,13 +20,19 @@ Metadata.prototype.update = function(updatedMetadata, cb) {
   var body = {};
   body[this.type] = updatedMetadata;
 
-  request
+  return new Promise(function(res, rej){
+    request
     .patch(fullUrl)
     .set('Authorization', 'Bearer ' + this.accessToken)
     .send(body)
-    .end(function(err, resp){
-      return cb(err, resp && resp.body);
-    });
+    .end(utils.responseHandler(cb || rej, function(resp){
+      if (cb) {
```

Comment:
This is going to be done in several places, extract it to a function (The check for cv and doing either resolve or calling the callback)